src/PartnerRegistrar.ts

```diff
@@ -113,7 +172,7 @@ export class PartnerRegistrar {
    * @param amount the amount for the name registration
    * @param addr the address to set for the name resolution
    */
-  async register (label: string, owner: string, secret: string, duration: BigNumber, amount: BigNumber, addr?: string): Promise<boolean> {
+  registerOp (...args: RegisterArgs): OperationResult<boolean> {
```

Comment:
the operator functions should be private and the arguments should be clearly stated